Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleDefinition/MSFT_IntuneRoleDefinition.psm1

```diff
@@ -0,0 +1,1179 @@
+                -ConnectionMode $ConnectionMode `
+                -ModulePath $PSScriptRoot `
+                -Results $Results `
+                -Credential $Credential
+
+            $dscContent += $currentDSCBlock
+            Save-M365DSCPartialExport -Content $currentDSCBlock `
+                -FileName $Global:PartialExportFileName
+            $i++
+            Write-Host $Global:M365DSCEmojiGreenCheckMark
+        }
+        return $dscContent
+    }
+    catch
+    {
+        Write-Host $Global:M365DSCEmojiGreenCheckMark
+        try
+        {
+            Write-Verbose -Message $_
+            $tenantIdValue = ''
+            if (-not [System.String]::IsNullOrEmpty($TenantId))
+            {
+                $tenantIdValue = $TenantId
+            }
+            elseif ($null -ne $Credential)
+            {
+                $tenantIdValue = $Credential.UserName.Split('@')[1]
+            }
+            Add-M365DSCEvent -Message $_ -EntryType 'Error' `
+                -EventID 1 -Source $($MyInvocation.MyCommand.Source) `
+                -TenantId $tenantIdValue
+        }
+        catch
+        {
+            Write-Verbose -Message $_
+        }
+        return ''
+    }
+}
+
+function Get-M365DSCAdditionalProperties
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param(
+        [Parameter(Mandatory = 'true')]
+        [System.Collections.Hashtable]
+        $Properties
+    )
+
+    $additionalProperties = @(
+
+    )
+    $results = @{'@odata.type' = '#microsoft.graph.deviceAndAppManagementRoleDefinition' }
+    $cloneProperties = $Properties.clone()
+    foreach ($property in $cloneProperties.Keys)
+    {
+        if ($property -in ($additionalProperties) )
+        {
+            $propertyName = $property[0].ToString().ToLower() + $property.Substring(1, $property.Length - 1)
+            if ($properties.$property -and $properties.$property.getType().FullName -like '*CIMInstance*')
+            {
+                if ($properties.$property.getType().FullName -like '*[[\]]')
+                {
+                    $array = @()
+                    foreach ($item in $properties.$property)
+                    {
+                        $array += Convert-M365DSCDRGComplexTypeToHashtable -ComplexObject $item
+
+                    }
+                    $propertyValue = $array
+                }
+                else
+                {
+                    $propertyValue = Convert-M365DSCDRGComplexTypeToHashtable -ComplexObject $properties.$property
+                }
+
+            }
+            else
+            {
+                $propertyValue = $properties.$property
+            }
+
+
+            $results.Add($propertyName, $propertyValue)
+
+        }
+    }
+    if ($results.Count -eq 1)
+    {
+        return $null
+    }
+    return $results
+}
+
+
+function Rename-M365DSCCimInstanceODataParameter
```

Comment:
Are all methods (in both resources) required? This resource doesn't deal with CIMInstances so I am questioning the need to have all these methods in here. Can we please double check and remove functions that are not required?